controllers/snapshot/snapshot_adapter.go

```diff
@@ -286,3 +322,21 @@ func (a *Adapter) getAllApplicationComponents(application *hasv1alpha1.Applicati
 
 	return &applicationComponents.Items, nil
 }
+
+// createIntegrationPipelineRun creates and returns a new integration PipelineRun. The Pipeline information and the parameters to it
+// will be extracted from the given integrationScenario. The integration's ApplicationSnapshot will also be passed to the
+// integration PipelineRun.
+func (a *Adapter) createIntegrationPipelineRun(application *hasv1alpha1.Application, integrationTestScenario *v1alpha1.IntegrationTestScenario, applicationSnapshot *appstudioshared.ApplicationSnapshot) error {
```

Comment:
We need to also append any labels and annotations that were on the build pipelinerun to the test pipelinerun for traceability of commits. So we may need to add a `getAllAnnotationsAndLabelsFromPipelineRun` function to the pipeline_adapter (since it has all the get functions to extract form the build pipeline). We could assign them to the ApplicationSnapshot and then extract them from the Snapshot to assign them to the Integration Pipeline as a simple solution

Reply:
I'm not sure whether we are getting any labels from build pipelinerun at the moment. In case we do not, does it make sense to have a story regarding that behaviour?

Reply:
Yes let's do a story. I will creat it. And add that functionality as part of PR support